sdk/communication/azure-communication-chat/src/test/java/com/azure/android/communication/chat/signaling/TrouterUtilsTest.java

```diff
@@ -0,0 +1,132 @@
+        assertEquals(CommunicationCloudEnvironment.PUBLIC, microsoftTeamsUserIdentifier.getCloudEnvironment());
+        assertEquals(false, microsoftTeamsUserIdentifier.isAnonymous());
+        assertEquals(USER_ID, microsoftTeamsUserIdentifier.getUserId());
+        assertEquals(teamsPublicUserMri, microsoftTeamsUserIdentifier.getRawId());
+    }
+
+    @Test
+    public void canParseTeamsDodUserMri() {
+        final String teamsDodUserMri = "8:dod:" + USER_ID;
+        CommunicationIdentifier communicationIdentifier = TrouterUtils.getCommunicationIdentifier(teamsDodUserMri);
+
+        assertNotNull(communicationIdentifier);
+        assertTrue(communicationIdentifier instanceof MicrosoftTeamsUserIdentifier);
+        MicrosoftTeamsUserIdentifier microsoftTeamsUserIdentifier =
+            (MicrosoftTeamsUserIdentifier) communicationIdentifier;
+        assertEquals(CommunicationCloudEnvironment.DOD, microsoftTeamsUserIdentifier.getCloudEnvironment());
+        assertFalse(microsoftTeamsUserIdentifier.isAnonymous());
+        assertEquals(USER_ID, microsoftTeamsUserIdentifier.getUserId());
+        assertEquals(teamsDodUserMri, microsoftTeamsUserIdentifier.getRawId());
+    }
+
+    @Test
+    public void canParseTeamsGcchUserMri() {
+        final String teamsGcchUserMri = "8:gcch:" + USER_ID;
+        CommunicationIdentifier communicationIdentifier = TrouterUtils.getCommunicationIdentifier(teamsGcchUserMri);
+
+        assertNotNull(communicationIdentifier);
+        assertTrue(communicationIdentifier instanceof MicrosoftTeamsUserIdentifier);
+        MicrosoftTeamsUserIdentifier microsoftTeamsUserIdentifier =
+            (MicrosoftTeamsUserIdentifier) communicationIdentifier;
+        assertEquals(CommunicationCloudEnvironment.GCCH, microsoftTeamsUserIdentifier.getCloudEnvironment());
+        assertFalse(microsoftTeamsUserIdentifier.isAnonymous());
+        assertEquals(USER_ID, microsoftTeamsUserIdentifier.getUserId());
+        assertEquals(teamsGcchUserMri, microsoftTeamsUserIdentifier.getRawId());
+    }
+
+    @Test
+    public void canParseTeamsVisitorUserMri() {
+        final String teamsVisitorUserMri = "8:teamsvisitor:" + USER_ID;
+        CommunicationIdentifier communicationIdentifier = TrouterUtils.getCommunicationIdentifier(teamsVisitorUserMri);
+
+        assertNotNull(communicationIdentifier);
+        assertTrue(communicationIdentifier instanceof MicrosoftTeamsUserIdentifier);
+        MicrosoftTeamsUserIdentifier microsoftTeamsUserIdentifier =
+            (MicrosoftTeamsUserIdentifier) communicationIdentifier;
+        assertEquals(CommunicationCloudEnvironment.PUBLIC, microsoftTeamsUserIdentifier.getCloudEnvironment());
+        assertTrue(microsoftTeamsUserIdentifier.isAnonymous());
+        assertEquals(USER_ID, microsoftTeamsUserIdentifier.getUserId());
+        assertEquals(teamsVisitorUserMri, microsoftTeamsUserIdentifier.getRawId());
+    }
+
+    @Test
+    public void canParseAcsUserMri() {
```

Comment:
instead of MRI, let's call it RawId. This should make sense for a developer that is not familiar with our internal terminolgy